src/integrations/BingAds/browser.js

```diff
@@ -67,34 +72,33 @@ class BingAds {
     Visit here(for details Parameter details): https://help.ads.microsoft.com/#apex/3/en/53056/2
     Under: What data does UET collect once I install it on my website?
     Updated syntax doc ref - https://help.ads.microsoft.com/#apex/ads/en/56916/2
+    Ecomm parameters ref - https://help.ads.microsoft.com/#apex/ads/en/60118/-1
   */
 
   track = (rudderElement) => {
-    const { type, properties, event } = rudderElement.message;
-    const { category, currency, value, revenue, total } = properties;
-    const eventToSend = type;
+    const { type, properties } = rudderElement.message;
+    const { eventAction } = properties;
+    const eventToSend = eventAction || type;
     if (!eventToSend) {
       logger.error('Event type not present');
       return;
     }
-    const payload = {
-      event_label: event,
+
+    let payload = {
+      ...buildCommonPayload(rudderElement.message),
+      ...buildEcommPayload(rudderElement.message),
     };
-    if (category) {
-      payload.event_category = category;
-    }
-    if (currency) {
-      payload.currency = currency;
-    }
-    if (value) {
-      payload.revenue_value = value;
-    }
-    if (revenue) {
-      payload.revenue_value = revenue;
-    }
-    if (total) {
-      payload.revenue_value = total;
-    }
+
+    let customProperties = {};
+    customProperties = extractCustomFields(
+      rudderElement.message,
+      customProperties,
+      ['properties'],
+      EXCLUSION_KEYS,
+    );
```

Comment:
```suggestion
    const customProperties = extractCustomFields(
      rudderElement.message,
      {},
      ['properties'],
      EXCLUSION_KEYS,
    );
```

Reply:
I see a bug in `extractCustomFields` function. [This line](https://github.com/rudderlabs/rudder-sdk-js/blob/3434f39b9b125a3693ae0f96c298f4d859961e34/src/utils/utils.js#L447) should be `[k]: get(messageContext, k),` instead of `k: get(messageContext, k),` (square brackets around k)

Reply:
Currently it's not affecting any destination. You can update this in another PR, but please keep a note of it.

Reply:
Incorporated customProperties suggestions. Will address the extractCustomFields utility change in a separate PR.